workflow/deterministic_wrappers.go

```diff
@@ -60,7 +60,8 @@ type (
 	WaitGroup = internal.WaitGroup
 )
 
-// Await blocks the calling thread until condition() returns true
+// Await blocks the calling thread until condition() returns true.
+// Do not mutate values or trigger side effects inside condition.
```

Comment:
Add the following:

The trigger is evaluated on every workflow state transition. Note that conditions that wait for time can be error-prone as nothing might cause evaluation. For example:

// workflow.Await(ctx, func() bool {
//   return workflow.Now() > someTime
// })

Might never return true unless some other event like Signal or activity completion would force the condition evaluation. For a time-based wait use workflow.AwaitWithTimeout function.